### PR TITLE
PLT-1789 Deleting posts with files now renames the file so that public links t…

### PR DESCRIPTION
**Pull request by jwilander**
…o those files no longer work

### CODE REVIEW

**Comment by coreyhulen** [comment]
Why can't we just check to see if the post is not deleted at https://github.com/mattermost/platform/blob/master/api/file.go#L397 vs the complexity of moving files and stuff.

**Comment by jwilander** [comment]
We don't have the PostId anywhere so we'd have to do a query on the Filename to get the post but there would need to be some guessing as the Filenames stored with the posts are really the end of URLs and not the filename that gets constructed in getFile()

**Comment by jwilander** [comment]
That would also introduce a dependency between files -> posts

**Comment by coreyhulen** [comment]
hmm...I thought postId was part of the file name.  Then can't a user override a file if the upload it with the same name?  or do we tack on a random uid?

**Comment by jwilander** [comment]
Yeah, we tack on a random UID

**Comment by jwilander** [comment]
Kind of makes me wonder if we should start tracking files in the DB..

File: api/file.go [hunk 1]
```diff
@@ -547,6 +547,41 @@ func writeFile(f []byte, path string) *model.AppError {
...
+		}
+
+		var auth aws.Auth
+		auth.AccessKey = utils.Cfg.FileSettings.AmazonS3AccessKeyId
+		auth.SecretKey = utils.Cfg.FileSettings.AmazonS3SecretAccessKey
+
+		s := s3.New(auth, awsRegion())
+		bucket := s.Bucket(utils.Cfg.FileSettings.AmazonS3Bucket)
+
+		if err := bucket.Del(oldPath); err != nil {
```
Comment (coreyhulen) [inline api/file.go]: Should delete after the move vs before?

**Comment by hmhealey** [comment]
+1 I definitely think tracking files in the DB would be the right way to go. That way we can get rid of some of my stuff that detects file size, format, etc through an API call and just store it in the DB

**Comment by coreyhulen** [comment]
+1
Please file a dev driven tix to refactor moving this into the db.

**Pull request merged**